File: 2006Scape Server/plugins/plugin/buttons/ButtonClick.java
package plugin.buttons;

import com.rs2.event.EventContext;
import com.rs2.event.EventSubscriber;
import com.rs2.event.impl.ButtonActionEvent;
import com.rs2.game.players.Player;

public abstract class ButtonClick implements EventSubscriber<ButtonActionEvent> {

    @Override
    public void subscribe(EventContext context, Player player, ButtonActionEvent event) {
        execute(player, event);
    }

    protected abstract void execute(Player player, ButtonActionEvent event);

    public abstract boolean test(ButtonActionEvent event);
}

File: 2006Scape Server/plugins/plugin/buttons/ButtonClick.kt
package plugin.buttons

import com.rs2.event.EventContext
import com.rs2.event.EventSubscriber
import com.rs2.event.impl.ButtonActionEvent
import com.rs2.game.players.Player


abstract class ButtonClick : EventSubscriber<ButtonActionEvent> {

	override fun subscribe(context: EventContext, player: Player, event: ButtonActionEvent) {
		execute(player, event)
	}
	
	abstract fun execute(player : Player, event : ButtonActionEvent);

}